### PR TITLE
update tests to run against 0.3.9

### PR DESCRIPTION
**Pull request by jbenet**
this updates some tests. however, the dht test is now failing and i cannot fix it. @whyrusleeping what changed?

### CODE REVIEW

File: test/tests.js [hunk 1]
```diff
@@ -601,17 +601,17 @@ describe('IPFS Node.js API wrapper tests', function () {
           }
 
           var result = [{
-            Ref: 'QmaMTzaGBmdLrispnPRTESta4yDQdK4uKSVcQez2No4h6q QmaeuuKLHzirbVoTjb3659fyyV381amjaGrU2pecHEWPrN add.js\n',
```
Comment (diasdavid) [inline test/tests.js]: no more `\n`, nice!

File: test/tests.js [hunk 2]
```diff
@@ -140,7 +140,7 @@ describe('IPFS Node.js API wrapper tests', function () {
           throw err
         }
         var added = res[res.length - 1]
-        assert.equal(added.Hash, 'QmaMTzaGBmdLrispnPRTESta4yDQdK4uKSVcQez2No4h6q')
+        assert.equal(added.Hash, 'QmZdsefMGMeG6bL719gX44XSVQrL6psEgRZdw1SGadFaK2')
```
Comment (Dignifiedquire) [inline test/tests.js]: How come this hash changed when upgrading?

Comment (jbenet) [inline test/tests.js]: because the output changed since 0.3.7-- the output of add now has one last hash which is the root of them all.

**Comment by diasdavid** [comment]
for reference: https://github.com/ipfs/go-ipfs/issues/1923

**Comment by diasdavid** [comment]
Important context https://botbot.me/freenode/ipfs/2015-11-02/?msg=53251219&page=2

**Comment by diasdavid** [comment]
https://botbot.me/freenode/ipfs/2015-11-02/?msg=53251438&page=2

```
06:44 <@whyrusleeping> so the puts work just fine
06:44 <@whyrusleeping> but when get is run, it collects the values
06:45 <@whyrusleeping> and then fails when it tries to 'pick the best one'
06:48 <+daviddias> got it got it :)
06:48 <+daviddias> well, adding notes of that all over the place
06:48 <+daviddias> but will consider that since the rest of the functionality is working, specially ipns, that the thing is good for now
06:48 <+daviddias> so that we can move forward with things that depend on this js-ipfs-api update
06:49 <+daviddias> sounds good?
06:53 <@whyrusleeping> woooo!
06:53 <@whyrusleeping> sounds good to me
```

Ready to merge, sounds good @jbenet ?

**Comment by diasdavid** [comment]
image
circleCI is having some trouble today

**Comment by jbenet** [comment]
@diasdavid sounds good.

**Comment by diasdavid** [comment]
sweeet!

**Pull request merged**